### ISSUE TITLE
failures when building C++

### ISSUE THREAD

**Issue by CMB** [snikket-im/snikket-sdk#32]
While running `make cpp/output.dso, the build process dies with this error:

```
Error: In file included from ./src/snikket/jingle/_PeerConnection/CppByte_Impl_.cpp:9:
include/snikket/jingle/_PeerConnection/CppByte_Impl_.h:58:50: error: 'std::byte' has not been declared
   58 |                 static unsigned char toInt( std::byte this1);
      |                                                  ^~~~
./src/snikket/jingle/_PeerConnection/CppByte_Impl_.cpp:34:15: error: 'unsigned char snikket::jingle::_PeerConnection::CppByte_Impl__obj::toInt' is not a static data member of 'class snikket::jingle::_PeerConnection::CppByte_Impl__obj'
   34 | unsigned char CppByte_Impl__obj::toInt( std::byte this1){
      |               ^~~~~~~~~~~~~~~~~
./src/snikket/jingle/_PeerConnection/CppByte_Impl_.cpp:34:46: error: 'byte' is not a member of 'std'
   34 | unsigned char CppByte_Impl__obj::toInt( std::byte this1){
      |                                              ^~~~
./src/snikket/jingle/_PeerConnection/CppByte_Impl_.cpp:34:46: note: 'std::byte' is only available from C++17 onwards
```

**Comment by CMB** [comment]
Here are a few more details about my environment.
I'm trying to build this on an Alpine 3.20 system.  I've also tried on a system running Alpine Edge.  I've tried with both haxe 4.34 from the pull request linked in the snikket-sdk README, as well as haxe 4.36.
I'm not really sure which of my two compilers is being used to compile the .cpp file.  I have both clang and gcc: *Alpine clang version 17.0.6*, and *gcc version 13.2.1 20240309 (Alpine 13.2.1_git20240309)*.
And the command I'm running is `make cpp/output.dso`.  Seems like this should be a bog-standard make-based build.  No clue what's wrong or if I'm missing something.

For those following along at home, I'll note that the haxe PR against the aports repo currently doesn't build out of the box, do to some ocaml library churn.  It needs this patch:

```
--- a/haxe.opam
+++ b/haxe.opam
@@ -25,7 +25,7 @@
   "dune" {>= "1.11" & < "3.16"}
   "sedlex" {>= "2.0"}
   "xml-light"
-  "extlib" {>= "1.7.8"}
+  "extlib" {= "1.7.9"}
   "ptmap" {>= "2.0.0"}
   "sha"
   "camlp-streams"
```

Not directly relevant to this report, I know, but it's worth mentioning in case someone tries building haxe on Alpine and can't get it to work.
